### DISCUSSION TITLE
Anyway to hook into Fields filter to override it

### DISCUSSION THREAD

**Discussion by saqueib** [statamic-rad-pack/runway#383] [Help] [answered]
> Runway provides a "Fields" filter to filter records in the Control Panel based on blueprint fields.

I have runway resource which is using `users` table. I want to filter users by **Role**. currently I have a computed field to display the roles in list view. 

**User Model**
```php
public function getRoleAttribute()
{
    return $this->allRoles()->pluck('title')->join(', ');
}
```

and blueprint field in yaml
```yaml
- handle: role
   field:
     type: text
     display: Roles
     listable: true
     visibility: computed
```

its using pivot table to store user roles 

| id | user_id | role_id |
|----|---------|---------|
| 1  | 19      | user    |
| 2  | 4       | editor  |

So when I apply the filter its failing with following message

`Column not found: 1054 Unknown column 'role' in 'where clause'`

![Screenshot 2023-12-07 at 8 55 22 PM](https://github.com/duncanmcclean/runway/assets/6937224/f84c0a2f-bd66-4a00-8dc8-dde5303c86d4)

Problem is that I dont have a Role eloquent model so I cant build a belongsToMany relation on users model. I tried adding a method roles and returning collection but I got it must return relationship error.

How can i make it work, anyway I can hook into field filters logic??

**Comment by duncanmcclean** [comment, marked as answer]
It should probably filter out "computed" fields from that list.

Then, you can create your own filter for the Roles stuff.

**Reply by saqueib** [reply]
@duncanmcclean thanks, created a custom filter did the trick.

I have some more questions if you can help with.

1. How can I change the Create {Resource} button link?
2. How can I hide some fields on create or edit form? Tried setting hidden: true, invisible: true, visible: false but nothing works.
3. Anyway to highlight a row with paid status?
4. How to change the formatting of a field value, like last login should show diffForHuman instead of timestamp?

Thanks

**Reply by duncanmcclean** [reply]
> How can I change the Create {Resource} button link?

You can change `Create {resource}` to `New {resource}` (or something similar) by defining a translation:

```json
// lang/en.json

{
    "Create :resource": "New :resource"
}
```

However, it's not any more flexible than that (the create  buttons Statamic Core work the same way).

> How can I hide some fields on create or edit form? Tried setting hidden: true, invisible: true, visible: false but nothing works.

You can do `visibility: hidden` on each of the fields you want to hide.

> Anyway to highlight a row with paid status?

Sorry, I'm not quite sure I understand what you mean.... 

> How to change the formatting of a field value, like last login should show diffForHuman instead of timestamp?

You'd need to create a "computed" field for this, w/ the `diffForHumans` part as an accessor on your model.

**Reply by saqueib** [reply]
thanks @duncanmcclean

> Anyway to highlight a row with paid status? 

I want to apply some class on a row `<tr class="bg-green-200">` so I can apply some formatting.

Like this 
![Screenshot 2023-12-11 at 11 30 26 AM](https://github.com/duncanmcclean/runway/assets/6937224/623e8bad-a535-468c-be8e-7b82e1608c28)

**Reply by saqueib** [reply]
@duncanmcclean I am also unable to validate unique email using laravel rule, it works fine for creating but update always gives `This value has already been taken.`.

```yaml
-
  handle: email
  field:
    type: text
    display: Email
    validate:
      - required
      - email
      - unique:users,email,{id}
```

**Reply by duncanmcclean** [reply]
> I want to apply some class on a row <tr class="bg-green-200"> so I can apply some formatting.

There's nothing in Runway to do this (there's not in Core Statamic either). However, you could potentially write your own JavaScript & CSS to make this happen. 

You could ask on the Statamic Discord server to see if anyone else has done this before.

> I am also unable to validate unique email using laravel rule, it works fine for creating but update always gives This value has already been taken..

Are you able to open an issue for this and I'll take a look when I can.
